cmd/config-migrate.go

```diff
@@ -27,6 +28,39 @@ import (
 // DO NOT EDIT following message template, please open a github issue to discuss instead.
 var configMigrateMSGTemplate = "Configuration file %s migrated from version '%s' to '%s' successfully.\n"
 
+var errConfigOld = errors.New("configuration version is older than expected")
+
+// Check if the config version is the latest,
+// returns nil if no migration needed, otherwise
+// proceed to migrate.
+func checkConfigForMigrate() error {
+	configFile := getConfigFile()
+
+	type serverConfigVersion struct {
+		Version string
+	}
+
+	srvCfgVersion := &serverConfigVersion{}
+	_, err := quick.Load(configFile, srvCfgVersion)
+	if os.IsNotExist(err) {
+		// Shouldn't happen, proceed and fail while
+		// reading actual config file.
+		return nil
+	} else if err != nil {
+		// On any other file read errors, fail.
+		return err
+	}
+
+	// Config version of expected version, no need to migrate.
+	if srvCfgVersion.Version == v18 {
+		return nil
+	}
+
+	// Config is readable but older, proceed to migrate.
+	return errConfigOld
+}
+
+// Migrates all config versions from "1" to "18".
 func migrateConfig() error {
```

Comment:
If so, the PR title/description doesn't convey this.  However migration fix is not that complex
```go
func migrateConfig() error {
	version, err := quick.GetVersion(getConfigFile())
	if err != nil {
		return err
	}

	switch version {
	case "1":
		err = purgeV1()
	case "2":
		if err = migrateV2ToV3(); err == nil {
			fallthrough
		}
	case "3":
		if err = migrateV3ToV4(); err == nil {
			fallthrough
		}
	case "4":
		if err = migrateV4ToV5(); err == nil {
			fallthrough
		}
	case "5":
		if err = migrateV5ToV6(); err == nil {
			fallthrough
		}
	case "6":
		if err = migrateV6ToV7(); err == nil {
			fallthrough
		}
	case "7":
		if err = migrateV7ToV8(); err == nil {
			fallthrough
		}
	case "8":
		if err = migrateV8ToV9(); err == nil {
			fallthrough
		}
	case "9":
		if err = migrateV9ToV10(); err == nil {
			fallthrough
		}
	case "10":
		if err = migrateV10ToV11(); err == nil {
			fallthrough
		}
	case "11":
		if err = migrateV11ToV12(); err == nil {
			fallthrough
		}
	case "12":
		if err = migrateV12ToV13(); err == nil {
			fallthrough
		}
	case "13":
		if err = migrateV13ToV14(); err == nil {
			fallthrough
		}
	case "14":
		if err = migrateV14ToV15(); err == nil {
			fallthrough
		}
	case "15":
		if err = migrateV15ToV16(); err == nil {
			fallthrough
		}
	case "16":
		if err = migrateV16ToV17(); err == nil {
			fallthrough
		}
	case "17":
		if err = migrateV17ToV18(); err == nil {
			fallthrough
		}
	}

	return err
}
```